tests/utils.go

```diff
@@ -3321,9 +3321,15 @@ func UnfinishedVMIPodSelector(vmi *v1.VirtualMachineInstance) metav1.ListOptions
 	vmi, err = virtClient.VirtualMachineInstance(vmi.Namespace).Get(vmi.Name, &metav1.GetOptions{})
 	Expect(err).ToNot(HaveOccurred())
 
-	fieldSelector := fields.ParseSelectorOrDie(
-		"status.phase!=" + string(k8sv1.PodFailed) +
-			",status.phase!=" + string(k8sv1.PodSucceeded))
+	fieldSelectorStr := "status.phase!=" + string(k8sv1.PodFailed) +
+		",status.phase!=" + string(k8sv1.PodSucceeded)
+
+	if vmi.Status.NodeName != "" {
```

Comment:
@vladikr had to cope with the situation where we look up the VMI when it is not yet scheduled, but the pod got scheduled in the meantime.